contracts/C2.sol

```diff
@@ -64,6 +64,24 @@ contract C2 is ERC20, Ownable {
         isLocked = true;
     }
 
+    function transfer(address recipient, uint256 amount) public override isLive returns (bool) {
+        require(amount == this.balanceOf(_msgSender()), "Only transfers of all tokens are allowed");
```

Comment:
This might be very hard to satisfy if your balance has 18 decimals.

Reply:
Hence the transferAll function below. This is mainly just to override the `transfer` function of the IERC20